### DISCUSSION TITLE
1 year long time-lapse. Some technical questions.

### DISCUSSION THREAD

**Discussion by Zampa1985** [gopro/labs#994] [Q&A]
Hi, after my question of the last week, today I bought a HERO 12 for my very long time-lapse of one year.
I have some first questions I really need to answer before to install the cam.

- Can I use any kind of power supply/mobile charger 5.0V/2A?

- How many photos I should take every day? I was thinking to take 48 photos a day.

- For a good result and to not ruin/crash the cam, should I program Labs to turn off the camera between the shots? Or it's maybe better to never turn off the cam for all year long?

- How can I download my photos without touching the cam? Of course I can use Quik App, but I need the cam always on and a connection, the same to use the Cloud.

Thank you so much,
Andrea

**Comment by dnewman-gpsw** [comment]
1) Any USB C power source will work. Although some I better than others.  Use *TUSB=1 if the power is producing error messages.
2) Up to you. Are you doing daytime only, or 24 hours?
3) In you doing 24 hours, never turning the camera off is a straight forward option.
4) This is tricky one, this is hard.  It is best if once a month you swap the SD card.  Labs can script the uploads, but it puts the scripting at risk. Cloud upload can hangs, particularly on some slower networks, putting the timelapse at greater risk of stopping.

**Reply by Zampa1985** [reply]
Thank you for the quick answer!

1 - Yes, I had an error just after 10 minutes. How can I use that code? I'm not able to use codes. I generate QR with the QR code creator.
2 - Actually I'd like to do it just daytime, I'm trying to understand how to do it.
3 - If I will understand how to do it just daytime, I will turn off the cam during nightime.
4 - Ok, I will try to have a very strong case so I can take out the camera from it without changing too much the framing. It is not possibile to use the usb cable, already connected to the camera for charging, to download the shots on a laptop with Quick software?

**Reply by dnewman-gpsw** [reply]
1. What error? Was it insufficient power? Use any QR Code generator, it all just text.  Here is the QR https://gopro.github.io/labs/control/set/?cmd=*TUSB=1
2 & 3. Demo of a daily, day only timelapse https://gopro.github.io/labs/control/dailytl/
4. You can down via the mobile app.  The point is to visit the camera, to make sure it is still running.

**Reply by Zampa1985** [reply]
Hi, thank you so much for your answer.

1 - Yes! I tried to use the code you gave me and it seems it works. Just a question, if now I add a QR for the long-term time-lapse, the *TUSB=1 wont' get cancelled? Will it still work?
2/3 - Ok thanks. Actually I will not do a "real" time-lapse, but just some photo every hour that I will edit at the end with a video editor software like Premiere/After effects.
4 - That's great!

**Reply by dnewman-gpsw** [reply]
The * makes it a permanent setting.

**Reply by Zampa1985** [reply]
That's great! Thanks again!

- I'm doing a test in my garden and I tried to download photos using Quick Pro but without success because it says "impossibile to find your GoPro".
Is it maybe because the GoPro bluetooth is not active? I didn't find in the QR Code Creator the ability to activate the camera Bluetooth.

**Reply by dnewman-gpsw** [reply]
Yes, BLE is turned off.  This is by design, as the extended timelapse (shutdown and sleep) is the lowest power state to get the longest run time. So just turn on the camera, cancel the current script, connect Quik, and do your transfer.  Then rescan the QR code. It is not possible to have the script running while doing a transfer.

**Reply by VreekFonq** [reply]
Hi, sorry to chime in. We are in the process of doing this with three Hero 12 Blacks. Same hurdles to take.
Why is cycling of power necessary? Can't we just keep it on and not rebooting/sleeping between subsequent daytime shoots? Cycling power is more risky for crashes, isn't it? 
And as i understand it: we also want to transfer the images or videos from the Go Pro monthly. There is no other way then to stop/power cycle the GoPro by physically pressing the power button and hooking the GoPro upto a laptop or pc? Regards, Ramon

**Reply by dnewman-gpsw** [reply]
You use the camera continuously with external power.  This the normal way, and it can run forever.  The reason to use sleep/reboot is power, it way-way more efficient.  So the Labs solution is mainly for battery configurations.

As for reliability, I run a HERO12 this way for six months without an issue.

**Reply by VreekFonq** [reply]
Hi, thanks for your quik response. Problem now is, that when we run without battery and simulate a power loss by removing the USB-c cable, the camera 'restarts' and repairs the video time-lapse file. But it doesn't continue with the script it should be running.

**Reply by dnewman-gpsw** [reply]
Which is why you should run with an internal battery. 

I think that file repair issue maybe addressed in H13, and should be ported H12.  Needs testing

**Reply by Espusen** [reply]
> You use the camera continuously with external power. This the normal way, and it can run forever. The reason to use sleep/reboot is power, it way-way more efficient. So the Labs solution is mainly for battery configurations.
> 
> As for reliability, I run a HERO12 this way for six months without an issue.

Can I just ask if this camera that has been running for 6 months have had the TUSB=1 program running?

**Reply by dbower44022** [reply]
David, can you explain the precise issue that you were referencing when  you said "the file repair issue maybe addressed in H13, and should be ported to H12".

**Reply by dnewman-gpsw** [reply]
If you are recording video (not photos) and the external power fails (no internal battery,) the file will not be closed.  The next time the camera boots, it will file repair, this seemed to prevent the Labs script continuing.  Likely all fixed.

**Reply by dbower44022** [reply]
So to be clear...  On H12 and H13 with WAKE=2 enabled and when powered by USB without a battery:
1 - On a power failure while recording a video or time lapse video, the video file will be corrupted.
2 - When USB power is restored after the power failure, the camera will "Fix" the corrupt video file and then proceed normally with the boot operations.

That is good news if correct!

Thanks

**Reply by dnewman-gpsw** [reply]
That is all correct.
